packages/flutter/lib/src/material/animated_icons/animated_icons.dart

```diff
@@ -295,5 +299,3 @@ T _interpolate<T>(List<T> values, double progress, _Interpolator<T> interpolator
   final double t = targetIdx - lowIdx;
   return interpolator(values[lowIdx], values[highIdx], t);
 }
-
-typedef T _Interpolator<T>(T a, T b, double progress);
```

Comment:
This is a great example of why I don't think this is a helpful lint. "Interpolator" tells me what this is. `T Function(T a, T b, double progress)` just looks arbitrary.

It happens to be private here, but we could easily (well, @amirh would surely stop us, but aside from that :-P) make it public, at which point we'd add lots of documentation to this typedef.

Reply:
The lint only occurs on private typedefs that are only used once.

Here I agree you loose that this typedef is an `Interpolator` but IMHO this sort of documentation is still there in the signature of `_interpolate` with the name of the parameter (`interpolator`).

```dart
T _interpolate<T>(
  List<T> values,
  double progress,
  T Function(T a, T b, double progress) interpolator,
) {
```

Reply:
Yeah but in that version it's buried in syntax and hard to spot.